### PR TITLE
feat(world): prevent invalid namespace strings [M-05]

### PR DESCRIPTION
**Pull request by alvrs**

### CODE REVIEW

File: packages/world/src/requireNamespace.sol [hunk 1]
```diff
@@ -13,7 +15,22 @@ using WorldResourceIdInstance for ResourceId;
...
     revert IWorldErrors.World_InvalidResourceType(RESOURCE_NAMESPACE, resourceId, resourceId.toString());
   }
 }
+
+/**
+ * @notice Checks if a given namespace string is valid.
+ * @dev Reverts with IWorldErrors.World_InvalidNamespace if the namespace includes the reserved separator string ("__").
+ * @param namespace The namespace to verify.
+ */
+function requireValidCharacters(bytes14 namespace) pure {
```
Comment (holic) [inline packages/world/src/requireNamespace.sol:30]: can we combine these two into one function `requireValidNamespace`?

File: .changeset/tidy-stingrays-think.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+"@latticexyz/world": major
+---
+
+Namespaces are not allowed to contain double underscores ("\_\_") anymore, as this sequence of characters is used to [separate the namespace and function selector](https://github.com/latticexyz/mud/pull/2168) in namespaced systems.
+This is to prevent signature clashes of functions in different namespaces.
+
+(Example: If namespaces were allowed to contain this separator string, a function "function" in namespace "namespace**my" would result in the namespaced function selector "namespace**my**function",
+and would clash with a function "my**function" in namespace "namespace".)
```
Comment (alvrs) [inline .changeset/tidy-stingrays-think.md:9]:
```suggestion
and would clash with a function "my\_\_function" in namespace "namespace".)
```

File: packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol [hunk 3]
```diff
@@ -44,8 +44,8 @@ contract WorldRegistrationSystem is System, IWorldErrors, LimitedCallContext {
    * @param namespaceId The unique identifier for the new namespace
    */
   function registerNamespace(ResourceId namespaceId) public virtual onlyDelegatecall {
-    // Require namespace ID to be a valid namespace
-    requireNamespace(namespaceId);
+    // Require namespace to be a valid namespace ID
```
Comment (alvrs) [inline packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol:47]:
```suggestion
    // Require namespace ID to be valid
```

File: packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol [hunk 4]
```diff
@@ -299,8 +299,8 @@ contract WorldRegistrationSystem is System, IWorldErrors, LimitedCallContext {
     ResourceId delegationControlId,
     bytes memory initCallData
   ) public onlyDelegatecall {
-    // Require namespace ID to be a valid namespace
-    requireNamespace(namespaceId);
+    // Require namespace to be a valid namespace ID
```
Comment (alvrs) [inline packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol:302]:
```suggestion
    // Require namespace ID to be a valid namespace
```

File: packages/world/src/modules/core/implementations/AccessManagementSystem.sol [hunk 5]
```diff
@@ -53,8 +53,8 @@ contract AccessManagementSystem is System, LimitedCallContext {
    * @param newOwner The address to which ownership should be transferred.
    */
   function transferOwnership(ResourceId namespaceId, address newOwner) public virtual onlyDelegatecall {
-    // Require the namespace ID to be a valid namespace
-    requireNamespace(namespaceId);
+    // Require the namespace to be a valid namespace ID
```
Comment (alvrs) [inline packages/world/src/modules/core/implementations/AccessManagementSystem.sol:56]:
```suggestion
    // Require the namespace ID to be a valid namespace
```

**Pull request merged**